extensions/widgets/pushButton/pushButton.lcb

```diff
@@ -0,0 +1,504 @@
+   put pProperties["opacity"] into mBackgroundOpacity
+end handler
+
+public handler OnCreate() returns nothing
+   put "iOS" into mWidgetTheme
+   put "Button" into mButtonLabel
+   put color [0, 121/255, 1] into mTextColor
+   put color [1, 1, 1] into mBackgroundColor
+   put "Transparent" into mBackgroundOpacity
+
+   put false into mShowBorder
+   put color [0, 121/255, 1] into mDefaultColor
+   put (my height)*(17/29) into mIosFontSize
+   put (my height)*(14/36) into mAndroidFontSize
+end handler
+
+public handler OnPaint() returns nothing
+   updateVariables()
+
+   if mWidgetTheme is "iOS" then
+      drawIosButton()
+   else if mWidgetTheme is "Android(Raised)" then
+      drawAndroidRaisedButton()
+   else if mWidgetTheme is "Android(Flat)" then
+      drawAndroidFlatButton()
+   end if
+
+end handler
+
+private handler drawIosButton() returns nothing
+
+   -- make sure the color of the label text is not white
+   variable tColorString as String
+   put colorToString(mTextColor) into tColorString
+   checkTextColor(tColorString)
+
+   -- background
+   set the paint of this canvas to fetchPaint("background")
+   fill fetchPath("background") on this canvas
+
+   -- label
+   set the font of this canvas to fetchFont()
+   set the paint of this canvas to fetchPaint("label")
+   fill text mButtonLabel at center of my bounds on this canvas
+end handler
+
+private handler drawAndroidRaisedButton() returns nothing
+
+   -- check if we need to show a border based on the background color
+   variable tColorString as String
+   put colorToString(mBackgroundColor) into tColorString
+   checkBackgroundColor(tColorString)
+
+   -- shadow
+
+   -- background
+   set the paint of this canvas to fetchPaint("background")
+   fill fetchPath("background") on this canvas
+
+   -- border
+   if mShowBorder then
+      set the paint of this canvas to fetchPaint("border")
+      set the stroke width of this canvas to 0.5
+      stroke fetchPath("border") on this canvas
+   end if
+
+   -- label
+   variable tLabel as String
+   put the upper of mButtonLabel into tLabel
+
+   set the font of this canvas to fetchFont()
+   set the paint of this canvas to fetchPaint("label")
+   fill text tLabel at center of my bounds on this canvas
+end handler
+
+private handler drawAndroidFlatButton() returns nothing
+
+   -- make sure the color of the label text is not white
+   variable tColorString as String
+   put colorToString(mTextColor) into tColorString
+   checkTextColor(tColorString)
+
+   -- label
+   variable tLabel as String
+   put the upper of mButtonLabel into tLabel
+
+   set the font of this canvas to fetchFont()
+   set the paint of this canvas to fetchPaint("label")
+   fill text tLabel at center of my bounds on this canvas
+end handler
+
+private handler updateVariables() returns nothing
+   put (my height)*(17/29) into mIosFontSize
+   put (my height)*(14/36) into mAndroidFontSize
+end handler
```

Comment:
Yeah probably have separate numerator and denominator constants is a good plan. If we don't already have a bug report / feature request about evaluated constants then can you file one?